### PR TITLE
drop mirror: skip DropFlowWorkflow unless mirror cdc

### PR DESCRIPTION
**Pull request by serprex**

### CODE REVIEW

File: flow/cmd/handler.go [hunk 1]
```diff
@@ -319,56 +319,68 @@ func (h *FlowRequestHandler) ShutdownFlow(
...
-	workflowOptions := client.StartWorkflowOptions{
-		ID:        workflowID,
-		TaskQueue: h.peerflowTaskQueueID,
-		SearchAttributes: map[string]interface{}{
-			shared.MirrorNameSearchAttribute: req.FlowJobName,
-		},
-	}
-	dropFlowHandle, err := h.temporalClient.ExecuteWorkflow(ctx, workflowOptions, peerflow.DropFlowWorkflow, req)
+	var isCdc bool
+	err = h.pool.QueryRow(ctx, "select exists(select 1 from flows where name = $1 and query_string is null)",
```
Comment (heavycrystal) [inline flow/cmd/handler.go:323]: could use `h.isCDCFlow`

Comment (serprex) [reply flow/cmd/handler.go]: nice, fixed

File: flow/cmd/mirror_status.go [hunk 2]
```diff
@@ -347,18 +347,14 @@ func (h *FlowRequestHandler) getQRepConfigFromCatalog(ctx context.Context, flowJ
 }
 
 func (h *FlowRequestHandler) isCDCFlow(ctx context.Context, flowJobName string) (bool, error) {
-	var query pgtype.Text
-	err := h.pool.QueryRow(ctx, "SELECT query_string FROM flows WHERE name = $1", flowJobName).Scan(&query)
+	var isCdc bool
+	err := h.pool.QueryRow(ctx, "select exists(select * from flows where name = $1 and isnull(query_string, '') = '')",
```
Comment (heavycrystal) [inline flow/cmd/mirror_status.go:351]: nil: SQL formatting

Comment (heavycrystal) [inline flow/cmd/mirror_status.go:351]: does `isnull` exist in Postgres?

Comment (serprex) [reply flow/cmd/mirror_status.go:351]: Yes

Comment (serprex) [reply flow/cmd/mirror_status.go]: replaced with `coalesce`

Comment (serprex) [reply flow/cmd/mirror_status.go]: How would you format it?

Comment (heavycrystal) [reply flow/cmd/mirror_status.go]: `SELECT exists(SELECT * FROM flows WHERE name=$1 and coalesce(query_string, '')='')"`

basically capitalize SQL keywords and lowercase functions and identifiers

**Pull request merged**